src/app/interfaces/customer.ts

```diff
@@ -1,81 +1,77 @@
+import { UserRole } from "./User"
 
-
+//TODO refactor to match the db/Api
 export interface Address {
-  address_id?: number,
-  city: string,
+  id?: number,
+  city_name: string,
   street: string,
   house_number: string,
   postal_code: string,
   country: string,
 }
-export interface User extends Address {
-  user_id?: number,
-  role_id?: number,
+export interface User {
+  id?: number,
   first_name: string,
   last_name: string,
   birth_date: Date,
-  address_id?: number,
   email: string,
   phone_number: string,
-  password: string,
-  national_registry_number: string
+  national_registry_number: string,
+  roles?: UserRole[],
+  addresses?: Address[],
+  password?: string,
+  active?: boolean,
 }
+
 export interface Customer extends User {
-  customer_id: number,
-  customer_type:string,
-}
-export interface CustomerContract extends Customer
-{
-  UserID?: number,
-   customer_type: string,
-   first_name: string,
-   last_name: string,
-   ContractID?: number
+  [x: string]: any
+  customer_type: CustomerType
 }
-export interface CustomerExtend
-{
-  UserID?: number,
-  RoleID?: number,
-  first_name: string,
-  last_name: string,
-  BirthDate: Date,
-  AddressID: number,
-  Email: string,
-  PhoneNumber: string,
-  Password: string,
-  ContractID: number
+
+
+export interface Meter{
+  meter_type:MeterType,
+  //index_value: number,
 }
-export interface Estimation {
-  estimation_id?: number,
-  service_type: number,
+
+export interface EstimationRegistration {
+  service_type: ServiceType,
   address_id?: number,
-  building_type: number,
+  building_type: BuildingType,
   family_size: number,
-  equipments: string,
+  equipment: EquipmentType [],
   past_consumption: number,
-  estimated_consumption:number,
+  meters : Meter[],
 }
 
-export interface Meter{
-  meter_id?:number,
-  physical_id?:string,
-  index_id?:number;
-  contract_id?:number,
-  meter_type:string,
-  index_value: number,
-  date?:Date,
+export enum BuildingType{
+
+  'Apartment'= 0,
+  'Closed Building'= 1,
+  'Semi Detached' = 2,
+  'Open Building'= 3,
+
 }
 
-export interface EstimatedContract extends Estimation
+export enum ServiceType
 {
-  start_date:Date,
-  end_date:Date,
-  customer_type:string,
-  tariff_id?:number,
-  estimation_id?:number,
-  address_id:number,
-  service_type:number,
-  status:string
+  Electricity= 0,
+  Gas= 1,
```

Comment:
Not equivalent to the api
![image](https://user-images.githubusercontent.com/32861260/174101956-7c4636e6-8a92-458b-9451-3b0de0b2c264.png)

should be 1 and 2